src/accelerate/big_modeling.py

```diff
@@ -93,6 +93,13 @@ def init_on_device(device: torch.device, include_buffers: bool = False):
         tst = nn.Liner(100, 100)  # on `cuda` device
     ```
     """
+    if include_buffers and hasattr(device, "__enter__") and callable(getattr(device, "__enter__")):
```

Comment:
Here it would be better to use a version check on PyTorch.

Reply:
+ a `TODO` comment to remove the version check once older PyTorch versions are no longer supported.